File: packages/cogo-chat-sdk-flutter/lib/utils/editor.dart
import '../models/editor_context.dart';

class EditorUtils {
  static EditorContext buildEditorContext({
    List<String>? openFiles,
    String? activeFile,
    String? selection,
  }) {
    return EditorContext(
      openFiles: openFiles ?? const [],
      activeFile: activeFile,
      selection: selection,
    );
  }
}

File: packages/cogo-chat-sdk-ts/src/editor.ts
import type { EditorContext } from './types.js';

export type Selection = { path: string; range: [number, number] };

export function buildEditorContext(params: {
  openFiles?: string[];
  activeFile?: string;
  selection?: Selection;
}): EditorContext {
  const ctx: EditorContext = {};
  if (params.openFiles && params.openFiles.length) ctx.open_files = params.openFiles;
  if (params.activeFile) ctx.active_file = params.activeFile;
  if (params.selection) ctx.selection = params.selection;
  return ctx;
}


